src/models/stores/ui.ts

```diff
@@ -305,9 +305,8 @@ export const UIModel = types
      *
      * @param doc a non curriculum document
      */
-    openResourceDocument(doc: DocumentModelType) {
-      const navTab = getNavTabOfDocument(doc.type)  || "";
-
+    openResourceDocument(doc: DocumentModelType, isStudentWorkspaceDoc?: boolean) {
```

Comment:
I don't think `isStudentWorkspaceDoc` should be passed here. The function should figure out what to do just based on the document. If the logic that is in `commented-documents.tsx` was moved to a function in this file lets say it is just called `isStudentWorspaceDoc(doc)`, and the document was passed `getNavTabOfDocument` instead the `doc.type`. Then `getNavTabOfDocument` can call `isStudentWorspaceDoc(doc)`.

Reply:
I'm still a bit confused after reading this a few times, but correct me if I'm misinterpreting it. 

What would be the function definitions be? 
`openResourceDocument(doc: DocumentModelType)`  <--- so we take out `isStudentWorkspaceDoc` as an arg.


`isStudentWorkspaceDoc(doc) ` <--- don't we also need to pass in the user's id? Here's what I have so far. 

![image](https://github.com/concord-consortium/collaborative-learning/assets/521934/370daedb-b127-4090-8d4f-afc7754c3e29)

Reply:
And you are also suggesting then that `getNavTabOfDocument(docType: string, isStudentWorkspaceDoc?: boolean) ` should be changed to just `getNavTabOfDocument(doc: DocumentModelType)` correct?